src/core/LogSystem.hpp

```diff
@@ -1,60 +1,37 @@
 #pragma once
 
-#include <godot_cpp/classes/engine.hpp>
-#include <godot_cpp/classes/ref_counted.hpp>
-#include <godot_cpp/core/class_db.hpp>
-#include <godot_cpp/variant/array.hpp>
-#include <godot_cpp/variant/string.hpp>
-#include <godot_cpp/variant/utility_functions.hpp>
-#include <map>
-
-#ifdef LIBMASZYNA_DEBUG_ENABLED
-#if defined(_MSC_VER)
-// MSVC doesn't require special handling.
-#define DEBUG(msg, ...)                                                                                                \
-    UtilityFunctions::print_rich("[color=orange](debug) " + vformat(String(msg), Array::make(__VA_ARGS__)) + "[/color]")
-#elif defined(__cplusplus) && __cplusplus >= 202002L
-// C++20 with __VA_OPT__
-#define DEBUG(msg, ...)                                                                                                \
-    UtilityFunctions::print_rich(                                                                                      \
-            "[color=orange](debug) " + vformat(String(msg) __VA_OPT__(, ) Array::make(__VA_ARGS__)) + "[/color]")
-#else
-// Use a workaround for C++11 to C++17
-#define DEBUG(msg, ...)                                                                                                \
-    UtilityFunctions::print_rich("[color=orange](debug) " + vformat(String(msg), ##__VA_ARGS__) + "[/color]")
-#endif
-#else
-#define DEBUG(msg, ...)
-#endif
+#include <godot_cpp/classes/file_access.hpp>
+
+#include <godot_cpp/classes/logger.hpp>
 
 namespace godot {
 
-    class LogSystem : public RefCounted {
-            GDCLASS(LogSystem, RefCounted);
+    class LogSystem : public Logger {
+        GDCLASS(LogSystem, Logger)
+        Ref<FileAccess> file_access;
 
-        private:
+    private:
+        protected:
+            static void _bind_methods() {};
         public:
-            static const char *LOG_UPDATED_SIGNAL;
+            LogSystem();
+            ~LogSystem() override;
 
-            static LogSystem *get_instance() {
-                return dynamic_cast<LogSystem *>(godot::Engine::get_singleton()->get_singleton("LogSystem"));
-            }
+            void _log_message(const String &message, bool p_error) override;
```

Comment:
this method should not be public (starts with underscore), or should be named properly

Reply:
Might be protected i think...
It has to be an override and therefore march parent so I'm not sure moving it to private scope would work